torch/csrc/jit/codegen/cuda/kernel_ir.h

```diff
@@ -1061,20 +1058,45 @@ class TORCH_CUDA_CU_API Scope {
     return exprs_[i];
   }
 
+  // Insert expr before expression at pos
+  void insert(size_t pos, Expr* expr);
+
   // Insert expr before ref
   void insert_before(Expr* ref, Expr* expr);
 
   // Insert expr after ref
   void insert_after(Expr* ref, Expr* expr);
 
-  bool contains(Expr* expr) const;
+  void push_back(Expr* e) {
+    exprs_.push_back(e);
```

Comment:
Just to inform me:
Are there any invariants on order of expressions in Scope?

Reply:
The order of expressions in a Scope determines how they appear in the final code. They are ordered in a way respecting the dependencies of tensors.